librosa/feature/spectral.py

```diff
@@ -950,6 +953,136 @@ def chroma_cqt(y=None, sr=22050, C=None, hop_length=512, fmin=None,
+
+    C : np.ndarray [shape=(d, t)] [Optional]
+        a pre-computed constant-Q spectrogram
+
+    hop_length : int > 0
+        number of samples between successive chroma frames
+
+    fmin : float > 0
+        minimum frequency to analyze in the CQT.
+        Default: 'C2' ~ 32.7 Hz
+
+    norm : int > 0, +-np.inf, or None
+        Column-wise normalization of the chromagram.
+
+    threshold : float
+        Pre-normalization energy threshold.  Values below the
+        threshold are set to the unit vector.
+
+    tuning : float
+        Deviation (in cents) from A440 tuning
+
+    n_chroma : int > 0
+        Number of chroma bins to produce
+
+    n_octaves : int > 0
+        Number of octaves to analyze above `fmin`
+
+    window : None or np.ndarray
+        Optional window parameter to `filters.cq_to_chroma`
+
+    bins_per_octave : int > 0
+        Number of bins per octave in the CQT.
+        Default: matches `n_chroma`
+
+    cqt_mode : ['full', 'hybrid']
+        Constant-Q transform mode
+
+    win_len_smooth : int > 0
+        Length of temporal smoothing window.
+        Default: 41
+
+    downsample_smooth : int > 0
+        Factor for downsampling after temporal smoothing.
+        (Setting to 1 keeps original feature rate.)
+        Default: 10
+
+    Returns
+    -------
+    chroma_cens : np.ndarray [shape=(n_chroma, t)]
+        The output cens-chromagram
+
+    See Also
+    --------
+
+    Examples
+    --------
+
+    '''
+
+    cqt_func = {'full': cqt, 'hybrid': hybrid_cqt}
+
+    if bins_per_octave is None:
+        bins_per_octave = n_chroma
+
+    # Build the CQT if we don't have one already
+    if C is None:
+        C = np.abs(cqt_func[cqt_mode](y, sr=sr,
+                                      hop_length=hop_length,
+                                      fmin=fmin,
+                                      n_bins=n_octaves * bins_per_octave,
+                                      bins_per_octave=bins_per_octave,
+                                      tuning=tuning,
+                                      real=False))
+
+    chroma = chroma_cqt(C=C, norm=None,
+                        fmin=fmin,
+                        bins_per_octave=bins_per_octave,
+                        n_octaves=n_octaves, window=window)
+
+    # L1-Normalization
+    chroma = util.normalize(chroma, norm=1, axis=0,
+                            mode='fill_constant', threshold=threshold)
+
+    # Quantize amplitudes
+    quant_steps = [0.4, 0.2, 0.1, 0.05]
+    quant_weights = [0.25, 0.25, 0.25, 0.25]
+
+    chroma_quant = np.zeros_like(chroma)
+
+    for cur_quant_step_idx, cur_quant_step in enumerate(quant_steps):
+        chroma_quant += (chroma > cur_quant_step) * quant_weights[cur_quant_step_idx]
+
+    # Apply temporal smoothing
+    # hanning is tapered to 0, thus throwing away those samples
+    # taking a plus 2 to retain the wished smoothing size
+    win = scipy.signal.hanning(win_len_smooth + 2, sym=False)[1:-1]
```

Comment:
Since you're using this window for convolution, and the first/last entries are 0, is it necessary to  trim them out here?

If it doesn't hurt, I'd say leave them in, since the code is clearer and there shouldn't be a noticeable difference in efficiency.